tc/sgx/trusted_worker_manager/enclave/work_order_processor_wpe.cpp

```diff
@@ -0,0 +1,289 @@
+            jret != JSONSuccess, "failed to serialize result");
+
+        JSON_Object* result = json_object_get_object(resp, "result");
+        tcf::error::ThrowIfNull(result, "failed to serialize the result");
+
+        JsonSetStr(result, "workOrderId", work_order_id.c_str(),
+            "failed to serialize work order id");
+        JsonSetStr(result, "workloadId", workload_id.c_str(),
+            "failed to serialize workload id");
+        JsonSetStr(result, "workerId", worker_id.c_str(),
+            "failed to serialize worker id");
+        JsonSetStr(result, "requesterId", requester_id.c_str(),
+            "failed to serialize requester id");
+        JsonSetStr(result, "workerNonce", worker_nonce.c_str(),
+            "failed to serialize worker nonce");
+        JsonSetStr(result, "workerSignature", worker_signature.c_str(),
+            "failed to serialize worker signature");
+        // Pack verification key and verification key signature
+        // from preprocessed json
+        JsonSetStr(result, "verificationKey",
+            ByteArrayToBase64EncodedString(
+                wo_pre_proc_keys.verification_key).c_str(),
+            "failed to serialize verification key");
+        JsonSetStr(result, "verificationKeySignature",
+            ByteArrayToBase64EncodedString(
+                wo_pre_proc_keys.verification_key_signature).c_str(),
+            "failed to serialize verification key signature");
+
+        jret = json_object_set_value(result, "outData", json_value_init_array());
+        tcf::error::ThrowIf<tcf::error::RuntimeError>(
+            jret != JSONSuccess, "failed to serialize the result data");
+
+        JSON_Array* data_array = json_object_get_array(result, "outData");
+        tcf::error::ThrowIfNull(data_array,
+            "failed to serialize the dependency array");
+
+        for (auto out_data : data_items_out)
+            out_data.Pack(data_array);
+
+        // Serialize the resulting json
+        size_t serializedSize = json_serialization_size(resp_value);
+        ByteArray serialized_response;
+        serialized_response.resize(serializedSize);
+
+        jret = json_serialize_to_buffer(resp_value,
+            reinterpret_cast<char*>(&serialized_response[0]), serialized_response.size());
+        tcf::error::ThrowIf<tcf::error::RuntimeError>(
+            jret != JSONSuccess, "workorder response serialization failed");
+
+        return serialized_response;
+    }  // WorkOrderProcessorWPE::CreateJsonOutput
+
+    void WorkOrderProcessorWPE::ComputeSignature(
+        ByteArray& signing_key, ByteArray& message_hash) {
+
+        tcf::crypto::sig::PrivateKey priv_sig_key(
+            ByteArrayToStr(signing_key));
+        ByteArray sig = priv_sig_key.SignMessage(message_hash);
+        worker_signature = base64_encode(sig);
+    }  // WorkOrderProcessorWPE::ComputeSignature
+
+    ByteArray WorkOrderProcessorWPE::Process(EnclaveData* enclaveData, std::string json_str) {
```

Comment:
Use parent class `WorkOrderProcessor::Process()` function and override `virtual ExecuteWorkOrder() `

Reply:
Done